src/main/java/sirius/biz/storage/layer2/BlobSoftRefProperty.java

```diff
@@ -64,6 +87,122 @@ private BlobSoftRefProperty(@Nonnull EntityDescriptor descriptor,
         super(descriptor, accessPath, field);
     }
 
+    /**
+     * Returns the {@link EntityDescriptor} of the referenced entity.
+     *
+     * @return the referenced entity descriptor
+     */
+    public List<EntityDescriptor> getReferencedDescriptors() {
+        if (referencedDescriptors == null) {
+            if (blobSoftRef == null) {
+                throw new IllegalStateException("Schema not linked!");
+            }
+            referencedDescriptors.add(mixing.getDescriptor(MongoBlob.class));
+            referencedDescriptors.add(mixing.getDescriptor(SQLBlob.class));
+        }
+
+        return Collections.unmodifiableList(referencedDescriptors);
+    }
+
+    private BlobSoftRef getBlobSoftRef() {
+        if (blobSoftRef == null) {
+            try {
+                this.blobSoftRef = (BlobSoftRef) field.get(accessPath.apply(descriptor.getReferenceInstance()));
+            } catch (Exception e) {
+                throw Exceptions.handle()
+                                .to(Mixing.LOG)
+                                .error(e)
+                                .withSystemErrorMessage(
+                                        "Unable to obtain a BlobSoftRef object from entity ref field ('%s' in '%s'): %s (%s)",
+                                        getName(),
+                                        descriptor.getType().getName())
+                                .handle();
+            }
+        }
+
+        return blobSoftRef;
+    }
+
+    @Override
+    protected void link() {
+        super.link();
+
+        try {
+            BaseEntityRef.OnDelete deleteHandler = getBlobSoftRef().getDeleteHandler();
+
+            if (deleteHandler == BaseEntityRef.OnDelete.CASCADE) {
+                getReferencedDescriptors().forEach(entityDescriptor -> entityDescriptor.addCascadeDeleteHandler(this::onDeleteCascade));
+            } else if (deleteHandler == BaseEntityRef.OnDelete.SET_NULL) {
+                if (!isNullable()) {
+                    Mixing.LOG.WARN("Error in property %s of %s: The field is not marked as NullAllowed,"
+                                    + " therefore SET_NULL is not a valid delete handler!", this, getDescriptor());
+                }
+
+                getReferencedDescriptors().forEach(entityDescriptor -> entityDescriptor.addCascadeDeleteHandler(this::onDeleteSetNull));
+            } else if (deleteHandler == BaseEntityRef.OnDelete.REJECT) {
+                throw new IllegalArgumentException("BlobSoftRef references do not accept REJECT as deleteHandler.");
```

Comment:
I would actually throw the exception in the constructor of the ref so it gets thrown right as such a ref is introduced during development and not when a first deletion occurs (which might be during testing or in production)